### PR TITLE
Feature/join room logic

### PR DESCRIPTION
**Pull request by JFCowboy**
## Description

[//]: # (Please include a summary of the change and which issues is fixed. Please also include relevant motivation and context. List any dependencies that are required to change.)

- Update endpoints
- fix null pointers and Type not found exceptions
- Room Controller and services to handle Join room logic

## Trello link:

https://trello.com/c/7pliEbQS/<Edit-this>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist
[//]: # (Go over all the following points, and put an `x` in all the boxes that apply.)
[//]: # (If you're unsure about any of these, don't hesitate to ask.)

- [ ] Changes were successfully tested.
- [ ] Dependencies. All dependencies are deployed.

## How Has This Been Tested?

[//]: # (Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration)

Some test in https://github.com/Devathon-2024-team5/Preguntonic-backend/pull/9

Test after [6b0e53d](https://github.com/Devathon-2024-team5/Preguntonic-backend/pull/16/commits/6b0e53dd6b1db1ba0be5fbee9a7d6687dd202342)

image

```
$ _INFO_:Receive subscribed message from destination /room/ROOM954, content = MESSAGE
content-length:860
message-id:jcy1wmf2-5
subscription:sub-0
content-type:application/json
destination:/room/ROOM954
content-length:860

{"room":{"current_players":[{"playerId":"f59e1fe0-939b-4c5e-9d72-f47984f0b3d1","admin":false,"ipAddress":null,"avatar":"cowboy.jpg","playerName":"Juan","status":"IN_LOBBY_READY","score":0,"responded":false,"responseTime":0,"responseId":null,"readyForNextQuestion":false},{"playerId":"76c38f94-8cd3-4082-9d0a-9d799e496e56","admin":false,"ipAddress":null,"avatar":"pato.jpg","playerName":"player_name","status":"IN_LOBBY_READY","score":0,"responded":false,"responseTime":0,"responseId":null,"readyForNextQuestion":false},{"playerId":"f6ce60b5-68ef-4125-9903-c19b563d5cff","admin":false,"ipAddress":null,"avatar":"dog.jpg","playerName":"player_name2","status":"IN_LOBBY_READY","score":0,"responded":false,"responseTime":0,"responseId":null,"readyForNextQuestion":false}],"ready_players":3,"max_players":4,"room_code":"ROOM954","room_status":null},"event":"READY"}
$ _INFO_:send STOMP message, destination = /app/rooms/ROOM954/lobby/players/76c38f94-8cd3-4082-9d0a-9d799e496e56/unready, content = {     "playerId": "76c38f94-8cd3-4082-9d0a-9d799e496e56",     "playerName": "player_name",     "avatar": "pato.jpg" }, header = 
$ _INFO_:Receive subscribed message from destination /room/ROOM954, content = MESSAGE
content-length:864
message-id:jcy1wmf2-6
subscription:sub-0
content-type:application/json
destination:/room/ROOM954
content-length:864

{"room":{"current_players":[{"playerId":"f59e1fe0-939b-4c5e-9d72-f47984f0b3d1","admin":false,"ipAddress":null,"avatar":"cowboy.jpg","playerName":"Juan","status":"IN_LOBBY_READY","score":0,"responded":false,"responseTime":0,"responseId":null,"readyForNextQuestion":false},{"playerId":"76c38f94-8cd3-4082-9d0a-9d799e496e56","admin":false,"ipAddress":null,"avatar":"pato.jpg","playerName":"player_name","status":"IN_LOBBY_UNREADY","score":0,"responded":false,"responseTime":0,"responseId":null,"readyForNextQuestion":false},{"playerId":"f6ce60b5-68ef-4125-9903-c19b563d5cff","admin":false,"ipAddress":null,"avatar":"dog.jpg","playerName":"player_name2","status":"IN_LOBBY_READY","score":0,"responded":false,"responseTime":0,"responseId":null,"readyForNextQuestion":false}],"ready_players":2,"max_players":4,"room_code":"ROOM954","room_status":null},"event":"UNREADY"}
```

image

### CODE REVIEW

File: src/main/java/com/devathon/preguntonic/controllers/RoomController.java [hunk 1]
```diff
@@ -27,25 +27,30 @@
 public interface RoomController {
```
Comment (peenyaa7) [inline src/main/java/com/devathon/preguntonic/controllers/RoomController.java:29]:
```suggestion
@MessageMapping("/rooms/{roomId}/lobby")
public interface RoomController {
```

File: src/main/java/com/devathon/preguntonic/controllers/RoomController.java [hunk 2]
```diff
@@ -27,25 +27,30 @@
...
   @GetMapping("/{roomId}")
-  ResponseEntity<Room> getRoom(@PathVariable String roomId);
+  ResponseEntity<Room> getRoom(@PathVariable("roomId") String roomId);
 
+  @PostMapping("/{roomId}/players")
+  ResponseEntity<BasicPlayer> addPlayerRoom(
+      @PathVariable("roomId") String roomId, @RequestBody BasicPlayer player);
+
+  // Websocket endpoints
   @MessageMapping("/rooms/{roomId}/lobby/join")
```
Comment (peenyaa7) [inline src/main/java/com/devathon/preguntonic/controllers/RoomController.java]:
```suggestion
  @MessageMapping("/join")
```

File: src/main/java/com/devathon/preguntonic/controllers/RoomController.java [hunk 3]
```diff
@@ -27,25 +27,30 @@
...
-  ResponseEntity<LobbyEvent> joinRoom(@DestinationVariable String roomId, BasicPlayer player);
+  ResponseEntity<LobbyEvent> joinRoom(
+      @DestinationVariable("roomId") String roomId, BasicPlayer player);
 
   @MessageMapping("/rooms/{roomId}/lobby/players/{playerId}/ready")
```
Comment (peenyaa7) [inline src/main/java/com/devathon/preguntonic/controllers/RoomController.java]:
```suggestion
  @MessageMapping("/players/{playerId}/ready")
```

File: src/main/java/com/devathon/preguntonic/controllers/RoomController.java [hunk 4]
```diff
@@ -27,25 +27,30 @@
...
   ResponseEntity<LobbyEvent> playerReadyInRoom(
-      @DestinationVariable String roomId, @DestinationVariable int playerId);
+      @DestinationVariable("roomId") String roomId, @DestinationVariable("playerId") int playerId);
 
   @MessageMapping("/rooms/{roomId}/lobby/players/{playerId}/unready")
```
Comment (peenyaa7) [inline src/main/java/com/devathon/preguntonic/controllers/RoomController.java]:
```suggestion
  @MessageMapping("/players/{playerId}/unready")
```

File: src/main/java/com/devathon/preguntonic/controllers/RoomControllerImpl.java [hunk 5]
```diff
@@ -65,79 +74,72 @@ public ResponseEntity<Room> getRoom(final String roomId) {
...
-      log.info(ROOM_NOT_FOUND, roomId);
-      return ResponseEntity.notFound().build();
+    try {
+      roomService.changePlayerReadyStatus(roomId, playerId, true);
+    } catch (final InvalidParameterException e) {
+      log.warn(PLAYER_READY_ERROR_MSG, e);
+      return ResponseEntity.badRequest().build();
+    } catch (final Exception e) {
+      log.warn(PLAYER_READY_ERROR_MSG, e);
+      return ResponseEntity.internalServerError().build();
```
Comment (peenyaa7) [inline src/main/java/com/devathon/preguntonic/controllers/RoomControllerImpl.java:105]: WebSocket works with ResponseEntity?

Comment (JFCowboy) [reply src/main/java/com/devathon/preguntonic/controllers/RoomControllerImpl.java:105]: Really we can return anything I decide use this to have a wrapper and a well defined type of responses.

File: src/main/java/com/devathon/preguntonic/configuration/PreguntonicConfiguration.java
Comment (peenyaa7) [inline src/main/java/com/devathon/preguntonic/configuration/PreguntonicConfiguration.java:1]: 👌

**Review by peenyaa7 — APPROVED** [review]

**Review by Alejandro25AR — APPROVED** [review]

**Pull request merged**